src/cdk-experimental/selection/row-selection.ts

```diff
@@ -30,10 +30,17 @@ export class CdkRowSelection<T> {
 
   @Input('cdkRowSelectionIndex')
   get index(): number|undefined { return this._index; }
-  set index(index: number|undefined) { this._index = coerceNumberProperty(index); }
-  private _index?: number;
+  set index(index: number|undefined) { CdkRowSelection._setIndexInput(this, index); }
+  protected _index?: number;
 
   constructor(readonly _selection: CdkSelection<T>) {}
 
+  // This has been extracted to a util because of TS 4 and VE.
+  // View Engine doesn't support property rename inheritance.
+  // TS 4.0 doesn't allow properties to override accessors or vice-versa.
+  protected static _setIndexInput<T>(classInstance: CdkRowSelection<T>, value: number|undefined) {
+    classInstance._index = coerceNumberProperty(value);
+  }
```

Comment:
For any of these setters that are only one line (mainly the coerce functions), can we just duplicate the setter and add a TODO to clean it up when we remove ViewEngine? E.g.
```ts
set index(index: number|undefined) { 
  // TODO: when we remove support for ViewEngine, change this setter to an input
  // alias in the decorator metadata.
  this._index = coerceNumberProperty(value);
}
```
I think duplicating these simple setters is better than the extra level of indirection from a static method

Reply:
I was originally going to duplicate this but @devversion suggested the latter approach. @devversion thoughts?

Reply:
I chatted w/ @devversion and I'd prefer to duplicate the one-line getter/setters and leave any multi-line ones as separate methods

Reply:
Yeah, I think the primary goal was to avoid the duplication of the larger setters (as it was before; that was quite a reduction in code health and maintainability). I didn't feel strongly about the single-line setters, but wouldn't mind them not being duplicated.

Reply:
My initial argument was that the setter logic in the CDK could change, and we wouldn't want to go through and change all duplicated setters all the time. It's not obvious from the CDK side that setters are duplicated.

Reply:
True, but I think that's tolerable compared to the extra level of indirection